app/shared/src/Device.js

```diff
@@ -2,39 +2,55 @@
 
 import { Dimensions, Platform } from 'react-native';
 
-function renewDimensions() {
-  try {
-    return Dimensions.get('screen');
-  } catch (error) {
-    // screen is not available in the 'test' environment
-    return { width: 0, height: 0 };
-  }
-}
+type ScreenDimensions = {
+  width: number,
+  height: number,
+};
 
 export default {
+  dimensions: {
+    width: 0,
+    height: 0,
+  },
+  renewDimensions() {
+    try {
+      const { width, height } = this.dimensions;
+      if (width && height) {
+        return { width, height };
+      }
+      return Dimensions.get('screen');
+    } catch (error) {
+      // screen is not available in the 'test' environment
+      return { width: 0, height: 0 };
+    }
+  },
+  // React native is unable to read correct dimensions when mulit tasking iOS
+  // Allow root view to set current dimensions and use those instead
+  setDimensions(dimensions: ScreenDimensions) {
+    this.dimensions = dimensions;
+  },
   isPortrait() {
-    const { height, width } = renewDimensions();
+    const { height, width } = this.renewDimensions();
     return height >= width;
   },
   isLandscape() {
-    const { height, width } = renewDimensions();
+    const { height, width } = this.renewDimensions();
     return width >= height;
   },
   isTablet() {
-    if (Platform.OS === 'ios' && Platform.isPad) {
-      return true;
-    }
```

Comment:
Why did you remove this condition? Maybe we should refactor it a little bit more - tablet is still tablet even though it's in the multitasking mode and it's a mistake we are adjusting the design based this I think.

Reply:
Maybe, now isTable is used to decide if we should show map with hotel list, render map view icon in header etc. Maybe we should add another method to decide if we should show map with hotel list?

Reply:
The logic of showing the icon is wrong. It doesn't depend on the tablet/phone but on the available width. Let me think about it for a while. I am not very happy with this implementation yet... 🙂